kotlinx-coroutines-debug/build.gradle.kts

```
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2016-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+import com.github.jengelman.gradle.plugins.shadow.tasks.*
+
+apply(plugin = "com.github.johnrengelman.shadow")
+
+// apply plugin to use autocomplete for Kover DSL
+apply(plugin = "org.jetbrains.kotlinx.kover")
```

```suggestion
plugins {
   id("com.github.johnrengelman.shadow")
   id("org.jetbrains.kotlinx.kover") // apply plugin to use autocomplete for Kover DSL
}
```

I don't remember where I read this in the docs, but if I recall correctly, configuration optimisation can be done only if plugins are declared in the newer `plugins` block, instead of legacy `apply` block.